pkg/network/setup/netstat.go

```diff
@@ -282,8 +274,15 @@ func isGuestAgentIfaceOriginatedFromOldVirtLauncher(guestAgentInterface api.Inte
 
 func updateVMIIfaceStatusWithGuestAgentData(ifaceStatus *v1.VirtualMachineInstanceNetworkInterface, guestAgentIface api.InterfaceStatus) {
 	ifaceStatus.InterfaceName = guestAgentIface.InterfaceName
-	ifaceStatus.IP = guestAgentIface.Ip
-	ifaceStatus.IPs = guestAgentIface.IPs
+	// IP data from the Guest Agent overrides previous iface status information in the following cases:
+	// - No status IPs existed before, i.e. GA data is adding new information.
+	// - Status IPs exist, however, GA information does not include any IP.
+	// In other words, if IP data already existed in the status, GA IP data will not override it.
+	// However, in case GA does not include IP data, it will clear IP status data (guest is not reachable by any IP).
+	if ifaceStatus.IP == "" || guestAgentIface.Ip == "" {
+		ifaceStatus.IP = guestAgentIface.Ip
+		ifaceStatus.IPs = guestAgentIface.IPs
```

Comment:
@EdDev 

We  encounter such an issue on https://github.com/harvester/harvester/issues/5881

When one NIC on the guestOS was assgined with mutli IPs, say IPv4 and IPv6; the IPv6 was assigned fast than IPv4.

The observed issue:

On VMI status, after IPv6 was updated, the later DHCP requested IPv4 on guest OS was not updated to VMI.

Namely, the `guestAgentIface.IPs has more records than ifaceStatus.IPs`, the above code seems not updating GA to iface.

could you help double confirm, thanks.